app/square_size.go

```diff
@@ -20,8 +20,8 @@ func (app *App) GovSquareSizeUpperBound(ctx sdk.Context) int {
 
 	gmax := int(app.BlobKeeper.GovMaxSquareSize(ctx))
 	// perform a secondary check on the max square size.
-	if gmax > appconsts.SquareSizeUpperBound(app.AppVersion()) {
-		gmax = appconsts.SquareSizeUpperBound(app.AppVersion())
+	if gmax > appconsts.SquareSizeUpperBound(app.AppVersion(ctx)) {
+		gmax = appconsts.SquareSizeUpperBound(app.AppVersion(ctx))
 	}
 
 	return gmax
```

Comment:
I partially agree with this suggestion but I think it could be expanded to do something like:

```go
	govMax := int(app.BlobKeeper.GovMaxSquareSize(ctx))
	hardMax := appconsts.SquareSizeUpperBound(version)
	return min(govMax, hardMax)
```

^ note I haven't tried that code so it may not work exactly